jwst/datamodels/container.py

```diff
@@ -244,29 +276,47 @@ def from_asn(self, asn_data, asn_file_path=None):
             for member in asn_data['products'][0]['members']:
                 if any([x for x in self.asn_exptypes if re.match(member['exptype'],
                                                                  x, re.IGNORECASE)]):
-                    infiles.append([member['expname'], member['exptype']])
+                    infiles.append(member)
                     logger.debug('Files accepted for processing {}:'.format(member['expname']))
         else:
-            infiles = [[member['expname'], member['exptype']] for member
+            infiles = [member for member
                        in asn_data['products'][0]['members']]
 
         if asn_file_path:
             asn_dir = op.dirname(asn_file_path)
-            infiles = [[op.join(asn_dir, f[0]), f[1]] for f in infiles]
+        else:
+            asn_dir = ''
 
         # Only handle the specified number of members.
         if self.asn_n_members:
             sublist = infiles[:self.asn_n_members]
         else:
             sublist = infiles
         try:
-            for filepath, exptype in sublist:
-                if self._save_open:
+            for member in sublist:
+                filepath = member['expname']
+                update_model = any(attr in member for attr in RECOGNIZED_MEMBER_FIELDS)
+                if update_model or self._save_open:
                     m = datamodel_open(filepath, memmap=self._memmap)
-                    m.meta.asn.exptype = exptype
+                    m.meta.asn.exptype = member['exptype']
+                    for attr, val in member.items():
+                        if attr in RECOGNIZED_MEMBER_FIELDS:
+                            if attr == 'tweakreg_catalog':
+                                if val.strip():
+                                    val = op.join(asn_dir, val)
+                                else:
+                                    val = None
```

Comment:
> How about
> 
> ```
> for attr, val in member.items():
>     if attr == 'tweakreg_catalog' and attr == 'tweakreg_catalog':
>         val = op.join(asn_dir, val.strip())
> ```

Why same check twice?

Reply:
I think @mcara meant:
```
for attr, val in member.items():
    if attr in RECOGNIZED_MEMBER_FIELDS and attr == 'tweakreg_catalog':
        val = op.join(asn_dir, val.strip())
```

Reply:
No, this would not work. If `val` is an empty string, I do not want to assign to `tweakreg_catalog` the ASN's directory name. An empty string means the values is not set => `None`.